src/storage/txn/flow_controller.rs

```diff
@@ -66,21 +67,37 @@ enum Trend {
 ///
 /// Here is a brief flow showing where the mechanism works:
 /// grpc -> check should drop(discardable ratio) -> limiter -> async write to raftstore
-pub struct FlowController {
+
+pub trait FlowController {
```

Comment:
no need to introduce this level of abstraction. We can just use TabletFlowController, and consume there is only one tablet when dynamic region is disabled.

Reply:
> no need to introduce this level of abstraction. We can just use TabletFlowController, and assume there is only one tablet when dynamic region is disabled.

There's a performance overhead for TabletFlowController, which will need a Hashmap and the lock for each access. 
Also logically we simply cannot use the same implementation, because it will get the tablet by hashmap's lookup on region_id, which won't really work for single rocksdb version------unless introduce some ugly if/else logic for single rocksdb version. 
That's why I'd rather to have this abstraction to make the implementation cleaner for each version.